pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -230,7 +307,41 @@ func (s *Service) getOrCreateLoadBalancer(openStackCluster *infrav1.OpenStackClu
 	return lb, nil
 }
 
-func (s *Service) getOrCreateListener(openStackCluster *infrav1.OpenStackCluster, listenerName, lbID string, port int) (*listeners.Listener, error) {
+// reconcileAPILoadBalancerListener ensures that the listener on the given port exists and is configured correctly.
+func (s *Service) reconcileAPILoadBalancerListener(lb *loadbalancers.LoadBalancer, openStackCluster *infrav1.OpenStackCluster, clusterName string, port int) error {
```

Comment:
do we have unit test coverage for this one?

Reply:
no we don't but it's not an easy one so I've filled https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/1906 so we can get back to it at some point.